src/GafferScene/EvaluateLightLinks.cpp

```diff
@@ -142,19 +146,21 @@ IECore::ConstCompoundObjectPtr EvaluateLightLinks::computeAttributes( const Scen
 	{
 		result->members()[ m_lightLinkAttrName ] = expressionToLightNames( illuminationExpressionData->readable(), inPlug(), lightsSet );
 	}
-	else
+	else if( path.size() == 1 )
 	{
-		// \todo: this transformation should probably get cached?
 		ConstPathMatcherDataPtr defaultLightsSetData = inPlug()->set( m_defaultLightSetName );
 		const PathMatcher &defaultLightsSet = defaultLightsSetData->readable();
 
-		IECore::StringVectorDataPtr defaultLightNames = new IECore::StringVectorData();
-		std::vector<std::string> &defaultLightNamesWritable = defaultLightNames->writable();
+		if( defaultLightsSet != lightsSet )
+		{
+			IECore::StringVectorDataPtr defaultLightNames = new IECore::StringVectorData();
+			std::vector<std::string> &defaultLightNamesWritable = defaultLightNames->writable();
 
-		PathMatcher defaultLights = defaultLightsSet.intersection( lightsSet );
-		defaultLights.paths( defaultLightNamesWritable );
+			PathMatcher defaultLights = defaultLightsSet.intersection( lightsSet );
```

Comment:
Are you doing this intersection because you don't trust the user not to done something invalid with the defaultLightsSet? If we keep this intersection, we need to also include the hash for the `__lights` set in `hashAttributes()`, because we're using its value here. Would be good to include this in the unit tests...